### PR TITLE
Clean up auto-setup.sh

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Clean up `auto-setup.sh` and other scripts.

**Why?**
For better readability as was suggested by @dnr in #2495.

**How did you test it?**
Build and run auto-setup docker image locally.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

**Review by dnr — COMMENTED** [review]
Sorry, I know you asked me to review all these more thoroughly a while ago, so I'm doing it now :)

A few suggestions and one bug.

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 set -eu -o pipefail
 
-export BIND_ON_IP="${BIND_ON_IP:-$(hostname -i)}"
+: "${BIND_ON_IP:=$(hostname -i)}"
+export BIND_ON_IP
 
 if [[ "${BIND_ON_IP}" =~ ":" ]]; then
     # ipv6
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply docker/entrypoint.sh:9]: The sequence is important there. I want them to be processed in this sequence no matter on how they are passed in args.

File: docker/auto-setup.sh [hunk 2]
```diff
@@ -4,73 +4,73 @@ set -eux -o pipefail
...
 
-SKIP_ADD_CUSTOM_SEARCH_ATTRIBUTES="${SKIP_ADD_CUSTOM_SEARCH_ATTRIBUTES:-false}"
+: "${SKIP_ADD_CUSTOM_SEARCH_ATTRIBUTES:=false}"
 
 # === Main database functions ===
 
 validate_db_env() {
-    if [ "${DB}" == "mysql" ]; then
-        if [ -z "${MYSQL_SEEDS}" ]; then
+    if [[ ${DB} == mysql ]]; then
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply docker/auto-setup.sh:62]: Wow, there are `case` in bash.

**Comment by alexshtin** [comment]
image

File: docker/entrypoint.sh [hunk 3]
```diff
@@ -16,12 +16,12 @@ fi
 dockerize -template /etc/temporal/config/config_template.yaml:/etc/temporal/config/docker.yaml
 
 # Automatically setup Temporal Server (databases, Elasticsearch, default namespace) if "autosetup" is passed as an argument.
-for arg in "$@" ; do [[ ${arg} == "autosetup" ]] && /etc/temporal/auto-setup.sh && break ; done
+for arg in "$@" ; do [[ ${arg} == autosetup ]] && /etc/temporal/auto-setup.sh && break ; done
```
Comment (dnr) [inline docker/entrypoint.sh:19]: could do `for arg; do ...` here and next one too

**Review by dnr — APPROVED** [review]

**Pull request merged**